### ISSUE TITLE
About the S model pretrain

### ISSUE THREAD

**Issue by 1angmul2** [AILab-CVC/YOLO-World#114]
I pre-trained the S model using configs/pretrain/yolo_world_s_vlpan_bn_2e-3_100e_4x8gpus_obj365v1_goldg_train_lvis_minival.py and found a ~1% difference compared to the log in the repo. 
`

 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=300 catIds=all] = 0.164
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=300 catIds=all] = 0.234
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=300 catIds=all] = 0.174
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=     s | maxDets=300 catIds=all] = 0.101
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=     m | maxDets=300 catIds=all] = 0.218
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=     l | maxDets=300 catIds=all] = 0.323
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=300 catIds=  r] = 0.130
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=300 catIds=  c] = 0.141
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=300 catIds=  f] = 0.192
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=300 catIds=all] = 0.244
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=     s | maxDets=300 catIds=all] = 0.123
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=     m | maxDets=300 catIds=all] = 0.305
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=     l | maxDets=300 catIds=all] = 0.449
03/11 12:41:16 - mmengine - INFO - Epoch(test) [602/602]    lvis/bbox_AP: 0.1640  lvis/bbox_AP50: 0.2340  lvis/bbox_AP75: 0.1740  lvis/bbox_APs: 0.1010  lvis/bbox_APm: 0.2180  lvis/bbox_APl: 0.3230  lvis/bbox_APr: 0.1300  lvis/bbox_APc: 0.1410  lvis/bbox_APf: 0.1920  data_time: 0.0008  time: 0.3425
`
In fact, initially, the model failed to converge when I tried pre-training with PyTorch 1.9.1. However, after switching to PyTorch 1.13, I achieved the aforementioned results. 

While it's possible that the issue was due to the PyTorch version, I would like to obtain precise versions of other libraries before attempting to switch PyTorch versions again. Below are the versions of libraries I used:
`
Package                Version 

accelerate             0.27.2
addict                 2.4.0
albumentations         1.4.1
certifi                2024.2.2
charset-normalizer     3.3.2
chumpy                 0.70
coloredlogs            15.0.1
contourpy              1.1.1
cycler                 0.12.1
Cython                 3.0.9
debugpy                1.8.1
defusedxml             0.7.1
diffusers              0.26.3
easydict               1.13
filelock               3.13.1
flatbuffers            24.3.6
fonttools              4.49.0
fsspec                 2024.2.0
h5py                   3.10.0
huggingface-hub        0.21.4
humanfriendly          10.0
idna                   3.6
imageio                2.34.0
imgaug                 0.4.0
importlib-metadata     7.0.1
importlib_resources    6.1.2
joblib                 1.3.2
json-tricks            3.17.3
kiwisolver             1.4.5
lazy_loader            0.3
llvmlite               0.41.1
lvis                   0.5.3
markdown-it-py         3.0.0
matplotlib             3.7.5
mdurl                  0.1.2
mmcv                   2.0.1
mmdet                  3.2.0
mmengine               0.8.4
mmpose                 1.1.0
mmyolo                 0.6.0
mpmath                 1.3.0
munkres                1.1.4
networkx               3.1
ninja                  1.11.1.1
numba                  0.58.1
numpy                  1.24.4
onnx                   1.15.0
onnx-simplifier        0.4.36
onnxruntime            1.17.1
opencv-python          4.9.0.80
opencv-python-headless 4.9.0.80
packaging              23.2
pandas                 2.0.3
pillow                 10.2.0
pip                    23.3.1
platformdirs           4.2.0
prettytable            3.10.0
protobuf               4.25.3
psutil                 5.9.8
pycocotools            2.0.7
Pygments               2.17.2
pyparsing              3.1.2
python-dateutil        2.9.0.post0
pytz                   2024.1
PyWavelets             1.4.1
PyYAML                 6.0.1
regex                  2023.12.25
requests               2.31.0
rich                   13.7.1
safetensors            0.4.2
scikit-image           0.21.0
scikit-learn           1.3.2
scipy                  1.10.0
setuptools             68.2.2
shapely                2.0.3
six                    1.16.0
supervision            0.18.0
sympy                  1.12
termcolor              2.4.0
terminaltables         3.1.10
threadpoolctl          3.3.0
tifffile               2023.7.10
tokenizers             0.15.2
tomli                  2.0.1
torch                  1.13.1+cu116
torchvision            0.14.1+cu116
tqdm                   4.66.2
transformers           4.38.2
typing_extensions      4.10.0
tzdata                 2024.1
urllib3                2.2.1
wcwidth                0.2.13
wheel                  0.41.2
xtcocotools            1.14.3
yapf                   0.40.1
yolo_world             0.1.0
zipp                   3.17.0
`
Thank you for the excellent work!

**Comment by wondervictor** [comment]
Hi @1angmul2, I think the version of PyTorch will not affect the overall performance. Could you provide more details about the pre-training, such as the number of GPUs or the pre-training data?

**Comment by 1angmul2** [comment]
I used the same configuration as mentioned in the repository, including 32*V100 and the exact same dataset. I only modified the dataset path in the configuration file. Initially, when I tried using PyTorch 1.9.1, the training didn't converge at all. However, after switching directly to PyTorch 1.13, the training proceeded normally. 
Now, besides the PyTorch version, I suspect that the differences might be due to different versions of either mmdet or mmcv. Could you please provide me with more precise versions of the dependency libraries? I would like to retry with an environment that is completely identical.

**Comment by wondervictor** [comment]
Sure. Could you compare the training log? We have provide the complete training log of YOLO-World-S, you can find it in https://github.com/AILab-CVC/YOLO-World?tab=readme-ov-file#zero-shot-inference-on-lvis-dataset

**Comment by 1angmul2** [comment]
After carefully comparing the logs, I realized my mistake. It turns out that when I recently modified the "__base__" configuration file to use absolute paths, I mistakenly set it to 'yolov8n', which resulted in decreased performance. However, what's remarkable is that the performance of the nano model only decreased by ~1%, which is quite impressive. 
Thank you for your response!
